poppy/wfe.py

```diff
@@ -299,3 +300,343 @@ def get_opd(self, wave, units='meters'):
+        return opd.real
+    
+    @utils.quantity_input(wavelength=u.meter)
+    def get_Cn2(self, wavelength):
+        """ Returns the index-of-refraction structure constant (m^-2/3).
+        
+        Parameters
+        -----------------
+        wavelength : float
+            The wavelength (m).
+        
+        References
+        -------------------
+        B. J. Herman and L. A. Strugala, in Proc. SPIE,
+        edited by P. B. Ulrich and L. E. Wilson
+        (International Society for Optics and Photonics, 1990), pp. 183–192.
+        """
+        
+        if all(item is not None for item in [self.r0, self.dz]):
+            r0 = self.r0.to(u.m)
+            wavelength2 = wavelength.to(u.m)**2
+            return wavelength2/self.dz * (r0/0.185)**(-5.0/3.0)
+        elif self.Cn2 is not None:
+            return self.Cn2.to(u.m**(-2/3))
+    
+    def rand_symmetrized(self, npix, sign):
+        """ Returns a real-valued random number array of shape (npix, npix)
+        with the symmetry required for a turbulent phase screen.
+        
+        Parameters
+        -----------------
+        npix : int
+            Number of pixels.
+        
+        sign : int
+            Sign of mirror symmetry. Must be either +1 or -1.
+        
+        References
+        -------------------
+        Eq. (65) in J. A. Fleck Jr, J. R. Morris, and M. D. Feit,
+        Appl. Phys. 10, 129 (1976).
+        """
+        
+        sign = float(sign)
+        
+        # create Gaussian, zero-mean, unit variance random numbers
+        a = np.random.normal(size=(npix, npix))
+        
+        # apply required symmetry
+        a[0, int(npix/2)+1:npix] = sign*a[0, 1:int(npix/2)][::-1]
+        a[int(npix/2)+1:npix, 0] = sign*a[1:int(npix/2), 0][::-1]
+        a[int(npix/2)+1:npix, int(npix/2)+1:npix] = sign*np.rot90(a[1:int(npix/2), 1:int(npix/2)], 2)
+        a[int(npix/2)+1:npix, 1:int(npix/2)] = sign*np.rot90(a[1:int(npix/2), int(npix/2)+1:npix], 2)
+        
+        # remove any overall phase resulting from the zero-frequency component
+        a[0, 0] = 0.0
+        
+        return a
+    
+    def rand_turbulent(self, npix):
+        """ Returns a complex-valued random number array of shape (npix, npix)
+        with the symmetry required for a turbulent phase screen.
+        
+        Parameters
+        -----------------
+        npix : int
+            Number of pixels.
+        
+        References
+        -------------------
+        Eq. (63) in J. A. Fleck Jr, J. R. Morris, and M. D. Feit,
+        Appl. Phys. 10, 129 (1976).
+        """
+        
+        # create real-valued random numbers with required symmetry
+        a = self.rand_symmetrized(npix, 1)
+        b = self.rand_symmetrized(npix, -1)
+        
+        # create complex-valued random number with required variance
+        c = (a + 1j*b)/np.sqrt(2.0)
+        
+        return c
+    
+    def power_spectrum(self, wave, L0=None, l0=None, kind='Kolmogorov'):
```

Comment:
I worry a little about having two parameters named so similarly as `L0` and `l0`. Seems to make it easy to accidentally have typos or other minor mistakes that cause unintended problems. Would it be better to have more explicit names `outer_scale` and `inner_scale`?